maldoca/ole/oss_utils.cc

```diff
@@ -438,6 +439,72 @@ bool BufferToUtf8::ConvertEncodingBufferToUTF8String(absl::string_view input,
     return true;
   }
   const char* input_ptr = input.data();
+
+#if defined(_WIN32)
+  int mb_size = 0;  // size of the multi-byte string
+  int wc_size = 0;  // size of the wide-character string
+  std::unique_ptr<wchar_t[]> wc_data;
+  const wchar_t* wc_str;
+  bool is_already_utf16 = code_page_ == kUtf16LECodePage;
+
+  // No need to convert to UTF-16, if the string is already UTF-16 encoded.
+  if (!is_already_utf16) {
+    // Calculate output size in UTF-16.
+    wc_size =
+        MultiByteToWideChar(code_page_, 0, input_ptr, input.size(), NULL, 0);
+    DLOG(INFO) << "Output size in UTF-16: " << wc_size;
+    if (wc_size <= 0) {
+      LOG(ERROR) << "Error while calculating output size in UTF-16: "
+                 << GetLastError();
+      return false;
+    }
+
+    // Allocate memory for the temp UTF-16 output and convert input to UTF-16.
+    wc_data = std::make_unique<wchar_t[]>(wc_size);
+    wc_str = wc_data.get();
+    wc_size = MultiByteToWideChar(code_page_, 0, input_ptr, input.size(),
+                                  wc_data.get(), wc_size);
+    if (wc_size <= 0) {
+      LOG(ERROR) << "Error while converting to UTF-16: " << GetLastError();
+      return false;
+    }
+  } else {
+    wc_str = reinterpret_cast<const wchar_t*>(input.data());
+    // We're casting the input from char (1 byte) to wchar_t (2 bytes), so we
+    // also have to half the size (in wchar_t).
+    wc_size = input.size() / 2;
+  }
+
+  // Calculate output size in UTF-8.
+  mb_size =
+      WideCharToMultiByte(CP_UTF8, 0, wc_str, wc_size, NULL, 0, NULL, NULL);
+  DLOG(INFO) << "Output size in UTF-8: " << mb_size;
+  if (mb_size <= 0) {
+    LOG(ERROR) << "Error while calculating output size in UTF-8: "
+               << GetLastError();
+    return false;
+  }
+
+  // Allocate proper memory and convert input to UTF-8.
+  out_str->resize(mb_size);
+  mb_size = WideCharToMultiByte(CP_UTF8, 0, wc_str, wc_size,
+                                const_cast<char*>(out_str->data()), mb_size,
```

Comment:
have to remove the const.  (1) warns about modifying the data after removing const, however i guess in this case it is fine as we use the output string, right?

(1) https://stackoverflow.com/questions/29883327/is-it-safe-to-remove-const-via-const-cast-and-invoke-a-non-const-function-that-d

Reply:
actually you're using the T* data() https://en.cppreference.com/w/cpp/container/vector/data  So it should be non-const.  Not sure why/if the compiler chose the const char* data() since the signature calls for char*.  You can try something like

char* out_ptr = out_str->data();   

to force it to use  T* data() to rid of the const_cast. At the end of the day it's fine.

Reply:
ok just use &out_str[0] then. That should work

Reply:
as discussed, reverting to "&(*out_str)[0]" as string.data() always returns a const: http://www.cplusplus.com/reference/string/string/data/